src/LineageTree/lineageTree.py

```diff
@@ -48,12 +45,198 @@ def get_next_id(self):
         Returns:
             int: next authorized id
         """
+        if self.max_id == -1 and self.nodes:
+            self.max_id = sorted(self.nodes)[-1]
         if self.next_id == []:
             self.max_id += 1
             return self.max_id
         else:
             return self.next_id.pop()
 
+    def complete_lineage(self, nodes: [int, set] = None):
+        """If there are leaves on the tree that exist on earlier timepoints than the last
+        defined by (self.t_e) will modify the branch to finish on the last timepoint.
+
+        Args:
+            nodes (int,set], optional): _description_. Defaults to None.
+        """
+        if nodes is None:
+            nodes = set(self.roots)
+        elif isinstance(nodes, int):
+            nodes = {nodes}
+        for node in nodes:
+            sub = set(self.get_sub_tree(node))
+            specific_leaves = sub.intersection(self.leaves)
+            for leaf in specific_leaves:
+                self.add_branch(leaf, self.t_e - self.time[leaf], reverse=True)
+
+    ###TODO pos can be callable and stay motionless (copy the position of the succ node, use something like optical flow)
+    def add_branch(
+        self,
+        succ: int,
+        length: int,
+        move_timepoints: bool = True,
+        pos: [callable, None] = None,
+        reverse: bool = False,
+    ):
+        """Adds a branch of specific length to a node either as a successor or as a predecessor.
+        If it is placed on top of a tree all the nodes will move timepoints #length down.
+
+        Args:
+            succ (int): Id of the successor (predecessor if reverse is False)
```

Comment:
Yeah... I add points before the node succ, which for me makes sense if its on reverse it will add points after this one. I know I made it a bit too compicated (even I was confused at some point :) ) . I was thinking to make it 2 functions, one for making predecessors and one for successors.